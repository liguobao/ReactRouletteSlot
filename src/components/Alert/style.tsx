/*
 * @Author: wzi
 * @Date: 2018-10-01 10:23:08
 * @Last Modified by: wzi
 * @Last Modified time: 2019-01-08 16:39:04
 */

import styled from '@common/helper/styled-component';

type WrapperProps = {
    mask: boolean;
};
export const Wrapper = styled.div`
    position: absolute;
    top: 0;
    left: 0;
    right: 0;
    bottom: 0;
    z-index: 100;
    background: ${(props: WrapperProps) => (props.mask ? '#988d8d' : null)};
    display: flex;
    align-items: center;
`;
export const ContextWrapper = styled.div`
    width: 100%;
    text-align: center;
`;
export const ContentWrapper = styled.span`
    border-radius: 10px;
    color: #fff;
    background-color: rgba(58, 58, 58, 0.9);
    display: inline-block;
    padding: 20px;
`;
export const Title = styled.div``;
export const Detail = styled.div``;
