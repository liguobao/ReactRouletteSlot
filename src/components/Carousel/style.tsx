/*
 * @Author: wzi
 * @Date: 2018-07-19 16:51:24
 * @Last Modified by: wzi
 * @Last Modified time: 2018-09-17 18:08:51
 */

import styled from '@common/helper/styled-component';
import { Carousel } from 'antd-mobile';
import CustomIcon from '@components/CustomIcon';

export const Img = styled(CustomIcon)`
    font-size: 100px;
`;
export const CarouselWrapper = styled(Carousel)`
    background: white;
    overflow: hidden;
`;
export const Wrapper = styled.div`
    min-height: 185px;
    transition: height 0.6s;

    ul {
        transition: height 0.6s;
        text-align: center;

        li a {
            min-height: 185px;
            img {
                width: 100%;
                max-width: 100%;
            }
        }
    }
`;

export const Holder = styled.div`
    height: 185px !important;
    text-align: center;
`;
