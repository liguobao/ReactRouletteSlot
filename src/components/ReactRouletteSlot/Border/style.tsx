/*
 * @Author: wzi
 * @Date: 2018-11-29 17:09:30
 * @Last Modified by: wzi
 * @Last Modified time: 2018-11-30 16:46:30
 */

import styled, { css } from '@common/helper/styled-component';
const animation = css`
    @keyframes blueWinkle {
        0% {
            background: #4999da;
            box-shadow: 0px 0px 10px 1px #a8c0d4;
        }
        49% {
            background: #4999da;
            box-shadow: 0px 0px 10px 1px #a8c0d4;
        }
        50% {
            background: yellow;
            box-shadow: 0px 0px 10px 1px white;
        }
        100% {
            background: yellow;
            box-shadow: 0px 0px 10px 1px white;
        }
    }
    @keyframes yellowWinkle {
        0% {
            background: yellow;
            box-shadow: 0px 0px 10px 1px white;
        }
        49% {
            background: yellow;
            box-shadow: 0px 0px 10px 1px white;
        }
        50% {
            background: #4999da;
            box-shadow: 0px 0px 10px 1px #a8c0d4;
        }
        100% {
            background: #4999da;
            box-shadow: 0px 0px 10px 1px #a8c0d4;
        }
    }
`;
type Props = {
    height: number;
    width: number;
};
export const Border = styled.div`
    border-radius: 10px;
    border-collapse: separate;
    position: relative;
    width: ${(props: Props) =>
        props.width ? props.width + 40 + 'px' : 'calc(100% + 40px)'};
    height: ${(props: Props) =>
        props.height ? props.height + 40 + 'px' : 'calc(100% + 40px)'};
    margin: 0 auto;
    background: #fc5a38;
`;

type BallProps = {
    top?: number;
    left?: number;
    bottom?: number;
    right?: number;
    isOdd: boolean;
    isRun: boolean;
};
export const Ball = styled.div`
    ${animation};
    width: 10px;
    height: 10px;
    border-radius: 50%;
    position: absolute;
    top: ${(props: BallProps) => props.top}px;
    left: ${(props: BallProps) => props.left}px;
    bottom: ${(props: BallProps) => props.bottom}px;
    right: ${(props: BallProps) => props.right}px;

    background: ${(props: BallProps) =>
        props.isRun ? '' : props.isOdd ? '#4999da' : 'yellow'};
    animation: ${(props: BallProps) =>
        props.isRun
            ? props.isOdd
                ? 'blueWinkle 1s infinite alternate'
                : 'yellowWinkle 1s infinite alternate'
            : ''};
`;
