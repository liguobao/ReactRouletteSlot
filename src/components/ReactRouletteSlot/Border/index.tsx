/*
 * @Author: wzi
 * @Date: 2018-10-01 10:23:08
 * @Last Modified by: wzi
 * @Last Modified time: 2018-11-30 14:57:02
 */

import React from 'react';
import { Border, Ball } from './style';
import { shouldUpdate } from '@common/decorator/decorator';
enum BallPositionType {
    'top',
    'bottom',
    'left',
    'right',
}
type BoardProps = {
    children: JSX.Element;
    size?: number;
    row: number;
    col: number;
    isRun: boolean;
};
type BoardState = {};
@shouldUpdate({
    props: ['children', 'row', 'col', 'isRun'],
})
export default class Board extends React.Component<BoardProps, BoardState> {
    getTotalSize = () => this.props.size + 20 * 2;
    getTop = (index: number, type: BallPositionType) => {
        switch (type) {
            case BallPositionType.left:
            case BallPositionType.right:
                return (
                    ((this.getTotalSize() - 20) / this.props.col / 2) * index +
                    5
                );
            case BallPositionType.top:
                return 5;
            default:
                return null;
        }
    };
    getLeft = (index: number, type: BallPositionType) => {
        switch (type) {
            case BallPositionType.bottom:
            case BallPositionType.top:
                return (
                    ((this.getTotalSize() - 20) / this.props.row / 2) * index +
                    5
                );
            case BallPositionType.left:
                return 5;
            default:
                return null;
        }
    };

    getBottom = (_index: number, type: BallPositionType) =>
        type === BallPositionType.bottom ? 5 : null;
    getRight = (_index: number, type: BallPositionType) =>
        type === BallPositionType.right ? 5 : null;

    getBalls = (size: number, type: BallPositionType) => {
        const result = [];
        for (let value = 0; value < size; value++) {
            result.push(
                <Ball
                    top={this.getTop(value, type)}
                    left={this.getLeft(value, type)}
                    isOdd={value % 2 === 0}
                    bottom={this.getBottom(value, type)}
                    right={this.getRight(value, type)}
                    isRun={this.props.isRun}
                    key={value}
                />
            );
        }
        return result;
    };
    TopBalls = (size: number) => this.getBalls(size, BallPositionType.top);
    BottomBalls = (size: number) =>
        this.getBalls(size, BallPositionType.bottom);
    LeftBalls = (size: number) => this.getBalls(size, BallPositionType.left);
    RightBalls = (size: number) => this.getBalls(size, BallPositionType.right);
    Balls = () => {
        const hSize = this.props.row * 2 + 1;
        const vSize = this.props.col * 2 + 1;
        const result = [
            this.TopBalls(hSize),
            this.BottomBalls(hSize),
            this.LeftBalls(vSize),
            this.RightBalls(vSize),
        ];
        return <React.Fragment>{result}</React.Fragment>;
    };
    render() {
        return (
            <React.Fragment>
                <Border size={this.props.size}>
                    <this.Balls />
                    {this.props.children}
                </Border>
            </React.Fragment>
        );
    }
}
