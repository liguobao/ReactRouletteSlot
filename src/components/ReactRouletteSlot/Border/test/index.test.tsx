/*
 * @Author: wzi
 * @Date: 2018-01-31 16:40:02
 * @Last Modified by: wzi
 * @Last Modified time: 2018-04-27 15:32:26
 */
import React from 'react';
import Border from '@components/ReactRouletteSlot/Border';
import { mount } from 'enzyme';
import 'jest-styled-components';
import toJson from 'enzyme-to-json';
import { Ball } from '@components/ReactRouletteSlot/Border/style';
jest.useFakeTimers();
class Child extends React.Component {
    render() {
        return <div>测试子组件</div>;
    }
}
describe('Border', () => {
    test('整体', () => {
        const wrapper = mount(
            <Border row={4} col={4} isRun={false} width={300} height={300}>
                <Child />
            </Border>
        );
        expect(toJson(wrapper)).toMatchSnapshot();
        expect(wrapper.contains(<Child />)).toBeTruthy();
        expect(wrapper.find(Ball).length).toBe(36);
    });
    test('size 为3', () => {
        const wrapper = mount(
            <Border row={3} col={3} isRun={false} width={300} height={300}>
                <Child />
            </Border>
        );
        expect(toJson(wrapper)).toMatchSnapshot();
        expect(wrapper.find(Ball).length).toBe(28);
    });
    test('running', () => {
        const wrapper = mount(
            <Border row={3} col={3} isRun={true} width={300} height={300}>
                <Child />
            </Border>
        );

        jest.advanceTimersByTime(250);
        expect(toJson(wrapper)).toMatchSnapshot();
    });
});
