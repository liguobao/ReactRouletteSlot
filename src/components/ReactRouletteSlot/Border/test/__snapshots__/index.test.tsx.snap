// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Border running 1`] = `
.c0 {
  border-radius: 10px;
  border-collapse: separate;
  position: relative;
  width: calc(100% + 40px);
  height: calc(100% + 40px);
  margin: 0 auto;
  background: #fc5a38;
}

.c1 {
  width: 10px;
  height: 10px;
  border-radius: 50%;
  position: absolute;
  top: 5px;
  left: NaNpx;
  bottom: px;
  right: px;
  -webkit-animation: blueWinkle 1s infinite alternate;
  animation: blueWinkle 1s infinite alternate;
}

.c2 {
  width: 10px;
  height: 10px;
  border-radius: 50%;
  position: absolute;
  top: 5px;
  left: NaNpx;
  bottom: px;
  right: px;
  -webkit-animation: yellowWinkle 1s infinite alternate;
  animation: yellowWinkle 1s infinite alternate;
}

.c3 {
  width: 10px;
  height: 10px;
  border-radius: 50%;
  position: absolute;
  top: px;
  left: NaNpx;
  bottom: 5px;
  right: px;
  -webkit-animation: blueWinkle 1s infinite alternate;
  animation: blueWinkle 1s infinite alternate;
}

.c4 {
  width: 10px;
  height: 10px;
  border-radius: 50%;
  position: absolute;
  top: px;
  left: NaNpx;
  bottom: 5px;
  right: px;
  -webkit-animation: yellowWinkle 1s infinite alternate;
  animation: yellowWinkle 1s infinite alternate;
}

.c5 {
  width: 10px;
  height: 10px;
  border-radius: 50%;
  position: absolute;
  top: NaNpx;
  left: 5px;
  bottom: px;
  right: px;
  -webkit-animation: blueWinkle 1s infinite alternate;
  animation: blueWinkle 1s infinite alternate;
}

.c6 {
  width: 10px;
  height: 10px;
  border-radius: 50%;
  position: absolute;
  top: NaNpx;
  left: 5px;
  bottom: px;
  right: px;
  -webkit-animation: yellowWinkle 1s infinite alternate;
  animation: yellowWinkle 1s infinite alternate;
}

.c7 {
  width: 10px;
  height: 10px;
  border-radius: 50%;
  position: absolute;
  top: NaNpx;
  left: px;
  bottom: px;
  right: 5px;
  -webkit-animation: blueWinkle 1s infinite alternate;
  animation: blueWinkle 1s infinite alternate;
}

.c8 {
  width: 10px;
  height: 10px;
  border-radius: 50%;
  position: absolute;
  top: NaNpx;
  left: px;
  bottom: px;
  right: 5px;
  -webkit-animation: yellowWinkle 1s infinite alternate;
  animation: yellowWinkle 1s infinite alternate;
}

<Board
  col={3}
  isRun={true}
  row={3}
>
  <styled.div>
    <div
      className="c0"
    >
      <Component>
        <styled.div
          bottom={null}
          isOdd={true}
          isRun={true}
          key="0"
          left={NaN}
          right={null}
          top={5}
        >
          <div
            className="c1"
          />
        </styled.div>
        <styled.div
          bottom={null}
          isOdd={false}
          isRun={true}
          key="1"
          left={NaN}
          right={null}
          top={5}
        >
          <div
            className="c2"
          />
        </styled.div>
        <styled.div
          bottom={null}
          isOdd={true}
          isRun={true}
          key="2"
          left={NaN}
          right={null}
          top={5}
        >
          <div
            className="c1"
          />
        </styled.div>
        <styled.div
          bottom={null}
          isOdd={false}
          isRun={true}
          key="3"
          left={NaN}
          right={null}
          top={5}
        >
          <div
            className="c2"
          />
        </styled.div>
        <styled.div
          bottom={null}
          isOdd={true}
          isRun={true}
          key="4"
          left={NaN}
          right={null}
          top={5}
        >
          <div
            className="c1"
          />
        </styled.div>
        <styled.div
          bottom={null}
          isOdd={false}
          isRun={true}
          key="5"
          left={NaN}
          right={null}
          top={5}
        >
          <div
            className="c2"
          />
        </styled.div>
        <styled.div
          bottom={null}
          isOdd={true}
          isRun={true}
          key="6"
          left={NaN}
          right={null}
          top={5}
        >
          <div
            className="c1"
          />
        </styled.div>
        <styled.div
          bottom={5}
          isOdd={true}
          isRun={true}
          key="0"
          left={NaN}
          right={null}
          top={null}
        >
          <div
            className="c3"
          />
        </styled.div>
        <styled.div
          bottom={5}
          isOdd={false}
          isRun={true}
          key="1"
          left={NaN}
          right={null}
          top={null}
        >
          <div
            className="c4"
          />
        </styled.div>
        <styled.div
          bottom={5}
          isOdd={true}
          isRun={true}
          key="2"
          left={NaN}
          right={null}
          top={null}
        >
          <div
            className="c3"
          />
        </styled.div>
        <styled.div
          bottom={5}
          isOdd={false}
          isRun={true}
          key="3"
          left={NaN}
          right={null}
          top={null}
        >
          <div
            className="c4"
          />
        </styled.div>
        <styled.div
          bottom={5}
          isOdd={true}
          isRun={true}
          key="4"
          left={NaN}
          right={null}
          top={null}
        >
          <div
            className="c3"
          />
        </styled.div>
        <styled.div
          bottom={5}
          isOdd={false}
          isRun={true}
          key="5"
          left={NaN}
          right={null}
          top={null}
        >
          <div
            className="c4"
          />
        </styled.div>
        <styled.div
          bottom={5}
          isOdd={true}
          isRun={true}
          key="6"
          left={NaN}
          right={null}
          top={null}
        >
          <div
            className="c3"
          />
        </styled.div>
        <styled.div
          bottom={null}
          isOdd={true}
          isRun={true}
          key="0"
          left={5}
          right={null}
          top={NaN}
        >
          <div
            className="c5"
          />
        </styled.div>
        <styled.div
          bottom={null}
          isOdd={false}
          isRun={true}
          key="1"
          left={5}
          right={null}
          top={NaN}
        >
          <div
            className="c6"
          />
        </styled.div>
        <styled.div
          bottom={null}
          isOdd={true}
          isRun={true}
          key="2"
          left={5}
          right={null}
          top={NaN}
        >
          <div
            className="c5"
          />
        </styled.div>
        <styled.div
          bottom={null}
          isOdd={false}
          isRun={true}
          key="3"
          left={5}
          right={null}
          top={NaN}
        >
          <div
            className="c6"
          />
        </styled.div>
        <styled.div
          bottom={null}
          isOdd={true}
          isRun={true}
          key="4"
          left={5}
          right={null}
          top={NaN}
        >
          <div
            className="c5"
          />
        </styled.div>
        <styled.div
          bottom={null}
          isOdd={false}
          isRun={true}
          key="5"
          left={5}
          right={null}
          top={NaN}
        >
          <div
            className="c6"
          />
        </styled.div>
        <styled.div
          bottom={null}
          isOdd={true}
          isRun={true}
          key="6"
          left={5}
          right={null}
          top={NaN}
        >
          <div
            className="c5"
          />
        </styled.div>
        <styled.div
          bottom={null}
          isOdd={true}
          isRun={true}
          key="0"
          left={null}
          right={5}
          top={NaN}
        >
          <div
            className="c7"
          />
        </styled.div>
        <styled.div
          bottom={null}
          isOdd={false}
          isRun={true}
          key="1"
          left={null}
          right={5}
          top={NaN}
        >
          <div
            className="c8"
          />
        </styled.div>
        <styled.div
          bottom={null}
          isOdd={true}
          isRun={true}
          key="2"
          left={null}
          right={5}
          top={NaN}
        >
          <div
            className="c7"
          />
        </styled.div>
        <styled.div
          bottom={null}
          isOdd={false}
          isRun={true}
          key="3"
          left={null}
          right={5}
          top={NaN}
        >
          <div
            className="c8"
          />
        </styled.div>
        <styled.div
          bottom={null}
          isOdd={true}
          isRun={true}
          key="4"
          left={null}
          right={5}
          top={NaN}
        >
          <div
            className="c7"
          />
        </styled.div>
        <styled.div
          bottom={null}
          isOdd={false}
          isRun={true}
          key="5"
          left={null}
          right={5}
          top={NaN}
        >
          <div
            className="c8"
          />
        </styled.div>
        <styled.div
          bottom={null}
          isOdd={true}
          isRun={true}
          key="6"
          left={null}
          right={5}
          top={NaN}
        >
          <div
            className="c7"
          />
        </styled.div>
      </Component>
      <Child>
        <div>
          测试子组件
        </div>
      </Child>
    </div>
  </styled.div>
</Board>
`;

exports[`Border size 为3 1`] = `
.c0 {
  border-radius: 10px;
  border-collapse: separate;
  position: relative;
  width: calc(100% + 40px);
  height: calc(100% + 40px);
  margin: 0 auto;
  background: #fc5a38;
}

.c1 {
  width: 10px;
  height: 10px;
  border-radius: 50%;
  position: absolute;
  top: 5px;
  left: NaNpx;
  bottom: px;
  right: px;
  background: #4999da;
}

.c2 {
  width: 10px;
  height: 10px;
  border-radius: 50%;
  position: absolute;
  top: 5px;
  left: NaNpx;
  bottom: px;
  right: px;
  background: yellow;
}

.c3 {
  width: 10px;
  height: 10px;
  border-radius: 50%;
  position: absolute;
  top: px;
  left: NaNpx;
  bottom: 5px;
  right: px;
  background: #4999da;
}

.c4 {
  width: 10px;
  height: 10px;
  border-radius: 50%;
  position: absolute;
  top: px;
  left: NaNpx;
  bottom: 5px;
  right: px;
  background: yellow;
}

.c5 {
  width: 10px;
  height: 10px;
  border-radius: 50%;
  position: absolute;
  top: NaNpx;
  left: 5px;
  bottom: px;
  right: px;
  background: #4999da;
}

.c6 {
  width: 10px;
  height: 10px;
  border-radius: 50%;
  position: absolute;
  top: NaNpx;
  left: 5px;
  bottom: px;
  right: px;
  background: yellow;
}

.c7 {
  width: 10px;
  height: 10px;
  border-radius: 50%;
  position: absolute;
  top: NaNpx;
  left: px;
  bottom: px;
  right: 5px;
  background: #4999da;
}

.c8 {
  width: 10px;
  height: 10px;
  border-radius: 50%;
  position: absolute;
  top: NaNpx;
  left: px;
  bottom: px;
  right: 5px;
  background: yellow;
}

<Board
  col={3}
  isRun={false}
  row={3}
>
  <styled.div>
    <div
      className="c0"
    >
      <Component>
        <styled.div
          bottom={null}
          isOdd={true}
          isRun={false}
          key="0"
          left={NaN}
          right={null}
          top={5}
        >
          <div
            className="c1"
          />
        </styled.div>
        <styled.div
          bottom={null}
          isOdd={false}
          isRun={false}
          key="1"
          left={NaN}
          right={null}
          top={5}
        >
          <div
            className="c2"
          />
        </styled.div>
        <styled.div
          bottom={null}
          isOdd={true}
          isRun={false}
          key="2"
          left={NaN}
          right={null}
          top={5}
        >
          <div
            className="c1"
          />
        </styled.div>
        <styled.div
          bottom={null}
          isOdd={false}
          isRun={false}
          key="3"
          left={NaN}
          right={null}
          top={5}
        >
          <div
            className="c2"
          />
        </styled.div>
        <styled.div
          bottom={null}
          isOdd={true}
          isRun={false}
          key="4"
          left={NaN}
          right={null}
          top={5}
        >
          <div
            className="c1"
          />
        </styled.div>
        <styled.div
          bottom={null}
          isOdd={false}
          isRun={false}
          key="5"
          left={NaN}
          right={null}
          top={5}
        >
          <div
            className="c2"
          />
        </styled.div>
        <styled.div
          bottom={null}
          isOdd={true}
          isRun={false}
          key="6"
          left={NaN}
          right={null}
          top={5}
        >
          <div
            className="c1"
          />
        </styled.div>
        <styled.div
          bottom={5}
          isOdd={true}
          isRun={false}
          key="0"
          left={NaN}
          right={null}
          top={null}
        >
          <div
            className="c3"
          />
        </styled.div>
        <styled.div
          bottom={5}
          isOdd={false}
          isRun={false}
          key="1"
          left={NaN}
          right={null}
          top={null}
        >
          <div
            className="c4"
          />
        </styled.div>
        <styled.div
          bottom={5}
          isOdd={true}
          isRun={false}
          key="2"
          left={NaN}
          right={null}
          top={null}
        >
          <div
            className="c3"
          />
        </styled.div>
        <styled.div
          bottom={5}
          isOdd={false}
          isRun={false}
          key="3"
          left={NaN}
          right={null}
          top={null}
        >
          <div
            className="c4"
          />
        </styled.div>
        <styled.div
          bottom={5}
          isOdd={true}
          isRun={false}
          key="4"
          left={NaN}
          right={null}
          top={null}
        >
          <div
            className="c3"
          />
        </styled.div>
        <styled.div
          bottom={5}
          isOdd={false}
          isRun={false}
          key="5"
          left={NaN}
          right={null}
          top={null}
        >
          <div
            className="c4"
          />
        </styled.div>
        <styled.div
          bottom={5}
          isOdd={true}
          isRun={false}
          key="6"
          left={NaN}
          right={null}
          top={null}
        >
          <div
            className="c3"
          />
        </styled.div>
        <styled.div
          bottom={null}
          isOdd={true}
          isRun={false}
          key="0"
          left={5}
          right={null}
          top={NaN}
        >
          <div
            className="c5"
          />
        </styled.div>
        <styled.div
          bottom={null}
          isOdd={false}
          isRun={false}
          key="1"
          left={5}
          right={null}
          top={NaN}
        >
          <div
            className="c6"
          />
        </styled.div>
        <styled.div
          bottom={null}
          isOdd={true}
          isRun={false}
          key="2"
          left={5}
          right={null}
          top={NaN}
        >
          <div
            className="c5"
          />
        </styled.div>
        <styled.div
          bottom={null}
          isOdd={false}
          isRun={false}
          key="3"
          left={5}
          right={null}
          top={NaN}
        >
          <div
            className="c6"
          />
        </styled.div>
        <styled.div
          bottom={null}
          isOdd={true}
          isRun={false}
          key="4"
          left={5}
          right={null}
          top={NaN}
        >
          <div
            className="c5"
          />
        </styled.div>
        <styled.div
          bottom={null}
          isOdd={false}
          isRun={false}
          key="5"
          left={5}
          right={null}
          top={NaN}
        >
          <div
            className="c6"
          />
        </styled.div>
        <styled.div
          bottom={null}
          isOdd={true}
          isRun={false}
          key="6"
          left={5}
          right={null}
          top={NaN}
        >
          <div
            className="c5"
          />
        </styled.div>
        <styled.div
          bottom={null}
          isOdd={true}
          isRun={false}
          key="0"
          left={null}
          right={5}
          top={NaN}
        >
          <div
            className="c7"
          />
        </styled.div>
        <styled.div
          bottom={null}
          isOdd={false}
          isRun={false}
          key="1"
          left={null}
          right={5}
          top={NaN}
        >
          <div
            className="c8"
          />
        </styled.div>
        <styled.div
          bottom={null}
          isOdd={true}
          isRun={false}
          key="2"
          left={null}
          right={5}
          top={NaN}
        >
          <div
            className="c7"
          />
        </styled.div>
        <styled.div
          bottom={null}
          isOdd={false}
          isRun={false}
          key="3"
          left={null}
          right={5}
          top={NaN}
        >
          <div
            className="c8"
          />
        </styled.div>
        <styled.div
          bottom={null}
          isOdd={true}
          isRun={false}
          key="4"
          left={null}
          right={5}
          top={NaN}
        >
          <div
            className="c7"
          />
        </styled.div>
        <styled.div
          bottom={null}
          isOdd={false}
          isRun={false}
          key="5"
          left={null}
          right={5}
          top={NaN}
        >
          <div
            className="c8"
          />
        </styled.div>
        <styled.div
          bottom={null}
          isOdd={true}
          isRun={false}
          key="6"
          left={null}
          right={5}
          top={NaN}
        >
          <div
            className="c7"
          />
        </styled.div>
      </Component>
      <Child>
        <div>
          测试子组件
        </div>
      </Child>
    </div>
  </styled.div>
</Board>
`;

exports[`Border 整体 1`] = `
.c0 {
  border-radius: 10px;
  border-collapse: separate;
  position: relative;
  width: calc(100% + 40px);
  height: calc(100% + 40px);
  margin: 0 auto;
  background: #fc5a38;
}

.c1 {
  width: 10px;
  height: 10px;
  border-radius: 50%;
  position: absolute;
  top: 5px;
  left: NaNpx;
  bottom: px;
  right: px;
  background: #4999da;
}

.c2 {
  width: 10px;
  height: 10px;
  border-radius: 50%;
  position: absolute;
  top: 5px;
  left: NaNpx;
  bottom: px;
  right: px;
  background: yellow;
}

.c3 {
  width: 10px;
  height: 10px;
  border-radius: 50%;
  position: absolute;
  top: px;
  left: NaNpx;
  bottom: 5px;
  right: px;
  background: #4999da;
}

.c4 {
  width: 10px;
  height: 10px;
  border-radius: 50%;
  position: absolute;
  top: px;
  left: NaNpx;
  bottom: 5px;
  right: px;
  background: yellow;
}

.c5 {
  width: 10px;
  height: 10px;
  border-radius: 50%;
  position: absolute;
  top: NaNpx;
  left: 5px;
  bottom: px;
  right: px;
  background: #4999da;
}

.c6 {
  width: 10px;
  height: 10px;
  border-radius: 50%;
  position: absolute;
  top: NaNpx;
  left: 5px;
  bottom: px;
  right: px;
  background: yellow;
}

.c7 {
  width: 10px;
  height: 10px;
  border-radius: 50%;
  position: absolute;
  top: NaNpx;
  left: px;
  bottom: px;
  right: 5px;
  background: #4999da;
}

.c8 {
  width: 10px;
  height: 10px;
  border-radius: 50%;
  position: absolute;
  top: NaNpx;
  left: px;
  bottom: px;
  right: 5px;
  background: yellow;
}

<Board
  col={4}
  isRun={false}
  row={4}
>
  <styled.div>
    <div
      className="c0"
    >
      <Component>
        <styled.div
          bottom={null}
          isOdd={true}
          isRun={false}
          key="0"
          left={NaN}
          right={null}
          top={5}
        >
          <div
            className="c1"
          />
        </styled.div>
        <styled.div
          bottom={null}
          isOdd={false}
          isRun={false}
          key="1"
          left={NaN}
          right={null}
          top={5}
        >
          <div
            className="c2"
          />
        </styled.div>
        <styled.div
          bottom={null}
          isOdd={true}
          isRun={false}
          key="2"
          left={NaN}
          right={null}
          top={5}
        >
          <div
            className="c1"
          />
        </styled.div>
        <styled.div
          bottom={null}
          isOdd={false}
          isRun={false}
          key="3"
          left={NaN}
          right={null}
          top={5}
        >
          <div
            className="c2"
          />
        </styled.div>
        <styled.div
          bottom={null}
          isOdd={true}
          isRun={false}
          key="4"
          left={NaN}
          right={null}
          top={5}
        >
          <div
            className="c1"
          />
        </styled.div>
        <styled.div
          bottom={null}
          isOdd={false}
          isRun={false}
          key="5"
          left={NaN}
          right={null}
          top={5}
        >
          <div
            className="c2"
          />
        </styled.div>
        <styled.div
          bottom={null}
          isOdd={true}
          isRun={false}
          key="6"
          left={NaN}
          right={null}
          top={5}
        >
          <div
            className="c1"
          />
        </styled.div>
        <styled.div
          bottom={null}
          isOdd={false}
          isRun={false}
          key="7"
          left={NaN}
          right={null}
          top={5}
        >
          <div
            className="c2"
          />
        </styled.div>
        <styled.div
          bottom={null}
          isOdd={true}
          isRun={false}
          key="8"
          left={NaN}
          right={null}
          top={5}
        >
          <div
            className="c1"
          />
        </styled.div>
        <styled.div
          bottom={5}
          isOdd={true}
          isRun={false}
          key="0"
          left={NaN}
          right={null}
          top={null}
        >
          <div
            className="c3"
          />
        </styled.div>
        <styled.div
          bottom={5}
          isOdd={false}
          isRun={false}
          key="1"
          left={NaN}
          right={null}
          top={null}
        >
          <div
            className="c4"
          />
        </styled.div>
        <styled.div
          bottom={5}
          isOdd={true}
          isRun={false}
          key="2"
          left={NaN}
          right={null}
          top={null}
        >
          <div
            className="c3"
          />
        </styled.div>
        <styled.div
          bottom={5}
          isOdd={false}
          isRun={false}
          key="3"
          left={NaN}
          right={null}
          top={null}
        >
          <div
            className="c4"
          />
        </styled.div>
        <styled.div
          bottom={5}
          isOdd={true}
          isRun={false}
          key="4"
          left={NaN}
          right={null}
          top={null}
        >
          <div
            className="c3"
          />
        </styled.div>
        <styled.div
          bottom={5}
          isOdd={false}
          isRun={false}
          key="5"
          left={NaN}
          right={null}
          top={null}
        >
          <div
            className="c4"
          />
        </styled.div>
        <styled.div
          bottom={5}
          isOdd={true}
          isRun={false}
          key="6"
          left={NaN}
          right={null}
          top={null}
        >
          <div
            className="c3"
          />
        </styled.div>
        <styled.div
          bottom={5}
          isOdd={false}
          isRun={false}
          key="7"
          left={NaN}
          right={null}
          top={null}
        >
          <div
            className="c4"
          />
        </styled.div>
        <styled.div
          bottom={5}
          isOdd={true}
          isRun={false}
          key="8"
          left={NaN}
          right={null}
          top={null}
        >
          <div
            className="c3"
          />
        </styled.div>
        <styled.div
          bottom={null}
          isOdd={true}
          isRun={false}
          key="0"
          left={5}
          right={null}
          top={NaN}
        >
          <div
            className="c5"
          />
        </styled.div>
        <styled.div
          bottom={null}
          isOdd={false}
          isRun={false}
          key="1"
          left={5}
          right={null}
          top={NaN}
        >
          <div
            className="c6"
          />
        </styled.div>
        <styled.div
          bottom={null}
          isOdd={true}
          isRun={false}
          key="2"
          left={5}
          right={null}
          top={NaN}
        >
          <div
            className="c5"
          />
        </styled.div>
        <styled.div
          bottom={null}
          isOdd={false}
          isRun={false}
          key="3"
          left={5}
          right={null}
          top={NaN}
        >
          <div
            className="c6"
          />
        </styled.div>
        <styled.div
          bottom={null}
          isOdd={true}
          isRun={false}
          key="4"
          left={5}
          right={null}
          top={NaN}
        >
          <div
            className="c5"
          />
        </styled.div>
        <styled.div
          bottom={null}
          isOdd={false}
          isRun={false}
          key="5"
          left={5}
          right={null}
          top={NaN}
        >
          <div
            className="c6"
          />
        </styled.div>
        <styled.div
          bottom={null}
          isOdd={true}
          isRun={false}
          key="6"
          left={5}
          right={null}
          top={NaN}
        >
          <div
            className="c5"
          />
        </styled.div>
        <styled.div
          bottom={null}
          isOdd={false}
          isRun={false}
          key="7"
          left={5}
          right={null}
          top={NaN}
        >
          <div
            className="c6"
          />
        </styled.div>
        <styled.div
          bottom={null}
          isOdd={true}
          isRun={false}
          key="8"
          left={5}
          right={null}
          top={NaN}
        >
          <div
            className="c5"
          />
        </styled.div>
        <styled.div
          bottom={null}
          isOdd={true}
          isRun={false}
          key="0"
          left={null}
          right={5}
          top={NaN}
        >
          <div
            className="c7"
          />
        </styled.div>
        <styled.div
          bottom={null}
          isOdd={false}
          isRun={false}
          key="1"
          left={null}
          right={5}
          top={NaN}
        >
          <div
            className="c8"
          />
        </styled.div>
        <styled.div
          bottom={null}
          isOdd={true}
          isRun={false}
          key="2"
          left={null}
          right={5}
          top={NaN}
        >
          <div
            className="c7"
          />
        </styled.div>
        <styled.div
          bottom={null}
          isOdd={false}
          isRun={false}
          key="3"
          left={null}
          right={5}
          top={NaN}
        >
          <div
            className="c8"
          />
        </styled.div>
        <styled.div
          bottom={null}
          isOdd={true}
          isRun={false}
          key="4"
          left={null}
          right={5}
          top={NaN}
        >
          <div
            className="c7"
          />
        </styled.div>
        <styled.div
          bottom={null}
          isOdd={false}
          isRun={false}
          key="5"
          left={null}
          right={5}
          top={NaN}
        >
          <div
            className="c8"
          />
        </styled.div>
        <styled.div
          bottom={null}
          isOdd={true}
          isRun={false}
          key="6"
          left={null}
          right={5}
          top={NaN}
        >
          <div
            className="c7"
          />
        </styled.div>
        <styled.div
          bottom={null}
          isOdd={false}
          isRun={false}
          key="7"
          left={null}
          right={5}
          top={NaN}
        >
          <div
            className="c8"
          />
        </styled.div>
        <styled.div
          bottom={null}
          isOdd={true}
          isRun={false}
          key="8"
          left={null}
          right={5}
          top={NaN}
        >
          <div
            className="c7"
          />
        </styled.div>
      </Component>
      <Child>
        <div>
          测试子组件
        </div>
      </Child>
    </div>
  </styled.div>
</Board>
`;
